nucypher/blockchain/eth/events.py

```diff
@@ -100,13 +97,15 @@ def __init__(self,
             raise ValueError(f"Contract agent must be provided")
         if not event_name:
             raise ValueError(f"Event name must be provided")
+        if not from_block:
```

Comment:
It feels strange that you have to check that for an argument that's positional, and explicitly typed as `int`. Also in the only call to `ContractEventsThrottler.__init__()` this value comes from `estimate_block_number_for_period` which does not return `None` either (on a side note, it returns `BlockNumber`, so perhaps that's what the type should be here).

Same applies to the other two checks, really.